---
title: "PFAS 2 MGS analysis"
author: Your name here
date: '`r format(Sys.time(), "%Y-%m-%d %H:%M")`'
output: 
  html_document:
    code_folding: show
    theme: cerulean
    number_sections: true
    highlight: monochrome
    fig_width: 7
    fig_height: 4
    toc: true
    toc_float: true
editor_options: 
  chunk_output_type: console
---

  
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning=FALSE, message=F)
```


# Background

## Purpose

Information describing the experiment and the goals for analysis here.

## Experimental Methods

Information on how samples were prepared and sequenced here. For Example if using standard lab workflow:

gDNA was extracted from fecal pellets using a DNeasy 96 PowerSoil Pro QIAcube HT Kit with liquid handling performed by a Integra mini-96 1250µL multichannel and a QIAcube HTP extraction platform. Samples were normalized to 6 ng/µL with 6 µL was used as the input to a 1/5th scale Illumina DNA Prep protocol using unique dual indexes based on the Illumina Tagmentation Set A-D. The resulting library was balanced using an Illumina iSeq100 2x150 run before full sequencing on a paired-end 150 NovaSeq X 25B lane (Novogene USA). Samples were sequenced with 43,196,663±5,747,337 (mean±sd) reads. Strain abundances were determined using StrainR2 with default parameters as previously described (22). All input strains were detected in the input community administered to mice.


## Notes

***

# User parameters

```{r}
indir<-"/data/SequencingRuns/2025May05_MGS2/tmp_output/output" # directory with mgs pipeline output it in
prefix<-"PFAS2_" #only pull samples with this sample prefix (for mixed runs)
metadatafile<-"PFAS2_metadata.xlsx" # must have column SampleID and Group at minimum
```


***
  
# R Setup

## Libraries
  
Load your libraries here. Version information for writing manuscripts will be included below:
  
```{r}
library(tidyverse)
library(readxl)
library(qiime2R)
library(vegan)


dir.create("figures") # this will create a folder for figures/output that might be necessary
dir.create("processed_data")
theme_set(theme_q2r()) # this will ensure that figures are stored with appropriate fonts/aesthetics
sessionInfo()
```

## Citation Information

This will be automatically generated based on the libraries loaded.

```{r}
packages_in_use <- c( sessionInfo()$basePkgs, names( sessionInfo()$loadedOnly ) )
lapply( X=packages_in_use, FUN=citation) %>%
  print()
```

***
  
# Data Import

## Normalizers

```{r}
normalizers<-
  list.files(paste0(indir, "/microbecensus/"), pattern="\\microbecensus$", full.names=TRUE) %>%
  grep(prefix, ., value=TRUE) %>%
  lapply(function(x) read_delim(x, skip=10, delim=":\t", col_names = c("Metric","Value")) %>% mutate(SampleID=basename(x) %>% gsub("\\.microbecensus","", .))) %>%
  bind_rows() %>%
  pivot_wider(names_from = Metric, values_from = Value) %>%
  dplyr::select(SampleID, AverageGenomeSize=average_genome_size, TotalBases=total_bases, GenomeEquivalents=genome_equivalents) %>%
  left_join(
    list.files(paste0(indir, "/metaphlan/"), pattern="\\.metaphlan$", full.names=TRUE) %>% 
      lapply(function(x) readLines(x)[3] %>% tibble(SampleID=basename(x) %>% gsub("\\.metaphlan$","", .), Reads=.)) %>%
      bind_rows() %>%
      mutate(Reads=gsub(" reads processed","", Reads) %>% gsub("#","", .) %>% as.numeric())
  )

interactive_table(normalizers)
write_tsv(normalizers,"processed_data/normalizers.tsv")
xfun::embed_file("processed_data/normalizers.tsv")
```

## Metadata
  
```{r}
metadata<-read_excel(metadatafile) # Should contain at least two columns, SampleID with names matching the SampleIDs in the file names, and Group which describes at least one experimental variable.
interactive_table(metadata)
```

## Taxonomic Analysis

```{r}
Taxa<-list.files(paste0(indir, "/metaphlan/"), pattern="\\.metaphlan$", full.names=TRUE) %>% 
  grep(prefix, ., value=TRUE) %>%
  lapply(function(x) read_tsv(x, skip=4) %>% mutate(SampleID=basename(x) %>% gsub("\\.metaphlan$","", .)))

Taxa<-
  lapply(Taxa, function(x){
    x %>% 
      mutate(tmp=gsub("..+\\|","", `#clade_name`) %>% gsub("__..+","", .)) %>%
      mutate(TaxLevel=case_when(
        tmp=="UNCLASSIFIED"~"Unclassified",
        tmp=="k"~"Kingdom",
        tmp=="p"~"Phylum",
        tmp=="c"~"Class",
        tmp=="o"~"Order",
        tmp=="f"~"Family",
        tmp=="g"~"Genus",
        tmp=="s"~"Species",
        tmp=="t"~"Subspecies",
      )) %>%
      dplyr::select(SampleID, TaxLevel, Taxon=`#clade_name`, Relative_Abundance=relative_abundance) %>%
      group_by(TaxLevel) %>%
      mutate(Relative_Abundance_NoUnassigned=Relative_Abundance/sum(Relative_Abundance)*100) %>%
      ungroup()
  })

Taxa<-bind_rows(Taxa)

interactive_table(Taxa)
write_tsv(Taxa, "processed_data/taxonomic_abundances.tsv")
xfun::embed_file("processed_data/taxonomic_abundances.tsv")
```

## Organisms of interest

```{r, eval=TRUE}
Orgs<-c(
  "k__Bacteria|p__Firmicutes|c__Bacilli|o__Lactobacillales|f__Enterococcaceae|g__Enterococcus",
  "k__Bacteria|p__Firmicutes|c__Bacilli|o__Lactobacillales|f__Enterococcaceae|g__Enterococcus|s__Enterococcus_faecalis",           
  "k__Bacteria|p__Firmicutes|c__Bacilli|o__Lactobacillales|f__Enterococcaceae|g__Enterococcus|s__Enterococcus_faecium" 
)


Taxa %>%
  filter(Taxon %in% Orgs) %>%
  dplyr::select(-Relative_Abundance, -TaxLevel) %>%
  pivot_wider(names_from = Taxon, values_from = Relative_Abundance_NoUnassigned, values_fill = 0) %>%
  pivot_longer(!SampleID, names_to = "Taxon", values_to = "Abundance") %>%
  interactive_table()

Taxa %>%
  filter(Taxon %in% Orgs) %>%
  dplyr::select(-Relative_Abundance, -TaxLevel) %>%
  pivot_wider(names_from = Taxon, values_from = Relative_Abundance_NoUnassigned, values_fill = 0) %>%
  pivot_longer(!SampleID, names_to = "Taxon", values_to = "Abundance") %>%
  group_by(Taxon) %>%
  mutate(log2Abundance=log2(Abundance+(2/3*min_nonzero(Abundance)))) %>%
  left_join(metadata) %>%
  mutate(Taxon=gsub("..+\\|","", Taxon)) %>%
  ggplot(aes(x=Group, y=log2Abundance, fill=Group)) +
  geom_boxplot() +
  geom_point() +
  facet_wrap(~Taxon)

ggsave("figures/OrganismsOfInterest.pdf", height=7.5, width=10, useDingbats=F)


Taxa %>%
  filter(Taxon %in% Orgs) %>%
  dplyr::select(-Relative_Abundance, -TaxLevel) %>%
  pivot_wider(names_from = Taxon, values_from = Relative_Abundance_NoUnassigned, values_fill = 0) %>%
  pivot_longer(!SampleID, names_to = "Taxon", values_to = "Abundance") %>%
  group_by(Taxon) %>%
  mutate(log2Abundance=log2(Abundance+(2/3*min_nonzero(Abundance)))) %>%
  left_join(metadata) %>%
  group_by(Taxon) %>%
  do(
    t.test(log2Abundance~Group, data=.) %>%
      broom::tidy()
  ) %>%
  interactive_table()
```


## Gene Abundances

```{r}
GeneAbundances<-
  list.files(paste0(indir, "/humann/"), pattern="genefamilies\\.tsv$", full.names=TRUE) %>% 
  grep(prefix, ., value=TRUE) %>%
  lapply(function(x){ read_tsv(x) %>% 
      mutate(SampleID=basename(x) %>% gsub("_genefamilies\\.tsv$","", .)) %>%
      mutate(Type=if_else(grepl("\\|", `# Gene Family`), "Stratified","Unstratified")) %>%
      dplyr::select(SampleID, GeneFamily=`# Gene Family`, Type, RPK=2)
  })

GeneAbundances<-bind_rows(GeneAbundances)

GeneAbundances<-
  GeneAbundances %>%
  left_join(normalizers %>% dplyr::select(SampleID, Reads, GenomeEquivalents)) %>%
  mutate(RPKG=RPK/GenomeEquivalents, RPKM=RPK/(Reads/1e6)) %>%
  dplyr::select(-Reads, -GenomeEquivalents)

#interactive_table(GeneAbundances) # too large to embed
write_tsv(GeneAbundances, "processed_data/NormalizedGeneAbundances.tsv")
#xfun::embed_file("processed_data/NormalizedGeneAbundances.tsv") # file to large to embed
```

## Pathway Abundances

```{r}
PathwayAbundances<-
  list.files(paste0(indir, "/humann/"), pattern="pathabundance\\.tsv$", full.names=TRUE) %>% 
  grep(prefix, ., value=TRUE) %>%
  lapply(function(x){ read_tsv(x) %>% 
      mutate(SampleID=basename(x) %>% gsub("_pathabundance\\.tsv$","", .)) %>%
      mutate(Type=if_else(grepl("\\|", `# Pathway`), "Stratified","Unstratified")) %>%
      dplyr::select(SampleID, Pathway=`# Pathway`, Type, Abundance=2) %>%
      filter(Pathway!="UNMAPPED" & !grepl("UNINTEGRATED", Pathway)) %>%
      group_by(Type) %>%
      mutate(Relative_Abundance=Abundance/sum(Abundance)*100) %>%
      ungroup()
  })

PathwayAbundances<-bind_rows(PathwayAbundances)

interactive_table(PathwayAbundances)
write_tsv(PathwayAbundances, "processed_data/PathwayAbundances.tsv")
xfun::embed_file("processed_data/PathwayAbundances.tsv")
```


***
  
# Exploratory Data Analysis
  
## Taxonomic
  
```{r}
TaxTable<-
  Taxa %>%
  ungroup() %>%
  filter(TaxLevel=="Class") %>%
  dplyr::select(SampleID, Taxon, Relative_Abundance) %>%
  pivot_wider(names_from = "SampleID", values_from = "Relative_Abundance", values_fill = 0) %>%
  column_to_rownames("Taxon")

taxa_barplot(TaxTable, metadata, "Group")
ggsave("figures/Tax_barplot.pdf", height=7.5, width=10, useDingbats=F)

TaxTable<-
  Taxa %>%
  ungroup() %>%
  filter(TaxLevel=="Species") %>%
  dplyr::select(SampleID, Taxon, Relative_Abundance) %>%
  pivot_wider(names_from = "SampleID", values_from = "Relative_Abundance", values_fill = 0) %>%
  column_to_rownames("Taxon")

taxa_heatmap(TaxTable, metadata, "Group")
ggsave("figures/Tax_heatmap.pdf", height=7.5, width=10, useDingbats=F)
```

### Taxa Differential Abundance Analysis

```{r}
TaxTable<-
  Taxa %>%
  ungroup() %>%
  filter(TaxLevel=="Species") %>%
  dplyr::select(SampleID, Taxon, Relative_Abundance) %>%
  pivot_wider(names_from = "SampleID", values_from = "Relative_Abundance", values_fill = 0) %>%
  pivot_longer(!Taxon, names_to = "SampleID") %>%
  group_by(Taxon) %>%
  mutate(value=log2(value+(2/3*(min_nonzero(value))))) #do log2 of percent abundance after filling in at 2/3 of lowest detectable value

DiffTaxa<-
  TaxTable %>%
  left_join(metadata) %>%
  group_by(Taxon) %>%
  do(
    t.test(value~Group, data=.) %>%
      broom::tidy()
  ) %>%
  ungroup() %>%
  mutate(FDR=p.adjust(p.value)) %>%
  dplyr::select(Taxon, log2FC=estimate, Pvalue=p.value, FDR) %>%
  arrange(Pvalue)

DiffTaxa %>%
  interactive_table()

DiffTaxa %>%
  mutate(Significant=if_else(FDR<0.1, "*","ns")) %>%
  mutate(Label=if_else(Significant=="*", Taxon, "")) %>%
  ggplot(aes(x=log2FC, y=-log10(Pvalue), color=Significant, label=Label)) +
  geom_point(shape=16, alpha=0.7) +
  ggrepel::geom_text_repel(size=3) +
  scale_color_manual(values=c("indianred","grey70")) 

ggsave("figures/Tax_Volcano.pdf", height=7.5, width=10, useDingbats=F)
```

### Taxa Ordination

```{r}
TaxTable<-
  Taxa %>%
  ungroup() %>%
  filter(TaxLevel=="Species") %>%
  dplyr::select(SampleID, Taxon, Relative_Abundance) %>%
  pivot_wider(names_from = "SampleID", values_from = "Relative_Abundance", values_fill = 0) %>%
  column_to_rownames("Taxon")

TaxDists<-vegan::vegdist(t(TaxTable), method="bray")
TaxPC<-ape::pcoa(TaxDists)

TaxPC$vectors %>%
  as.data.frame() %>%
  rownames_to_column("SampleID") %>%
  left_join(metadata) %>%
  ggplot(aes(x=Axis.1, y=Axis.2, fill=Group, label=SampleID)) +
  geom_point(shape=21) +
  ggrepel::geom_text_repel(size=3) +
  xlab(paste0("PC1:", round(TaxPC$values$Rel_corr_eig[1]*100, 2), "%")) +
  ylab(paste0("PC2:", round(TaxPC$values$Rel_corr_eig[2]*100, 2), "%"))

ggsave("figures/Tax_BrayPCoA.pdf", height=7.5, width=10, useDingbats=F)

adonis2(TaxDists~Group, data=metadata[match(labels(TaxDists), metadata$SampleID),]) %>%
  print()
```

### Taxa Diversity

```{r}
TaxTable<-
  Taxa %>%
  ungroup() %>%
  filter(TaxLevel=="Species") %>%
  dplyr::select(SampleID, Taxon, Relative_Abundance) %>%
  pivot_wider(names_from = "SampleID", values_from = "Relative_Abundance", values_fill = 0) %>%
  column_to_rownames("Taxon")

AlphaDiversity<-
  data.frame(Shannon=vegan::diversity(TaxTable, "shannon", MARGIN=2)) %>% rownames_to_column("SampleID") %>%
  left_join(
    data.frame(Simpson=vegan::diversity(TaxTable, "simpson", MARGIN=2)) %>% rownames_to_column("SampleID")
  ) %>%
  left_join(
    data.frame(ObservedTaxa=vegan::specnumber(TaxTable, MARGIN=2)) %>% rownames_to_column("SampleID")
  )

interactive_table(AlphaDiversity)

AlphaDiversity %>%
  pivot_longer(!SampleID, names_to = "Metric", values_to = "Diversity") %>%
  left_join(metadata) %>%
  ggplot(aes(x=Group, y=Diversity, fill=Group)) +
  geom_boxplot(outlier.alpha = 0) +
  geom_jitter(height=0, width=0.2, shape=21) +
  facet_wrap(~Metric, scales="free")
ggsave("figures/Tax_diversity.pdf", height=7.5, width=10, useDingbats=F)


AlphaDiversity %>%
  pivot_longer(!SampleID, names_to = "Metric", values_to = "Diversity") %>%
  left_join(metadata) %>%
  group_by(Metric) %>%
  do(
    t.test(Diversity~Group, data=.) %>%
      broom::tidy()
  ) %>%
  interactive_table()
```


## Pathway Abundances

```{r}
Pathway_filled<-
  PathwayAbundances %>% 
  filter(Type=="Unstratified") %>%
  dplyr::select(SampleID, Pathway, Relative_Abundance) %>%
  pivot_wider(names_from = Pathway, values_from = Relative_Abundance, values_fill = 0) %>%
  column_to_rownames("SampleID") %>%
  t()

Pathway_filled %>%
  taxa_heatmap(., metadata = metadata, "Group")
ggsave("figures/Pathway_heatmap.pdf", height=7.5, width=10, useDingbats=F)


TaxPC<-ape::pcoa(vegan::vegdist(t(Pathway_filled), method = "bray", ))

TaxPC$vectors %>%
  as.data.frame() %>%
  rownames_to_column("SampleID") %>%
  left_join(metadata) %>%
  ggplot(aes(x=Axis.1, y=Axis.2, fill=Group, label=SampleID)) +
  geom_point(shape=21) +
  ggrepel::geom_text_repel(size=3) +
  xlab(paste0("PC1:", round(TaxPC$values$Rel_corr_eig[1]*100, 2), "%")) +
  ylab(paste0("PC2:", round(TaxPC$values$Rel_corr_eig[2]*100, 2), "%"))

ggsave("figures/Pathway_BrayPCoA.pdf", height=7.5, width=10, useDingbats=F)

adonis2(TaxDists~Group, data=metadata[match(labels(TaxDists), metadata$SampleID),]) %>%
  print()
```


### Pathway Differential Abundance

```{r}
PathTable<-
  Pathway_filled %>%
  as.data.frame() %>%
  rownames_to_column("Pathway") %>%
  pivot_longer(!Pathway, names_to = "SampleID") %>%
  group_by(Pathway) %>%
  mutate(value=log2(value+(2/3*(min_nonzero(value))))) #do log2 of percent abundance after filling in at 2/3 of lowest detectable value

DiffPath<-
  PathTable %>%
  left_join(metadata) %>%
  group_by(Pathway) %>%
  do(
    t.test(value~Group, data=.) %>%
      broom::tidy()
  ) %>%
  ungroup() %>%
  mutate(FDR=p.adjust(p.value)) %>%
  dplyr::select(Pathway, log2FC=estimate, Pvalue=p.value, FDR) %>%
  arrange(Pvalue)

DiffPath %>%
  interactive_table()

DiffPath %>%
  mutate(Significant=if_else(FDR<0.1, "*","ns")) %>%
  mutate(Label=if_else(Significant=="*", Pathway, "")) %>%
  ggplot(aes(x=log2FC, y=-log10(Pvalue), color=Significant, label=Label)) +
  geom_point(shape=16, alpha=0.7) +
  ggrepel::geom_text_repel(size=3) +
  scale_color_manual(values=c("indianred","grey70")) 

ggsave("figures/Pathway_Volcano.pdf", height=7.5, width=10, useDingbats=F)
```


## Gene Abundances

```{r}
GeneAbundances_filled<-
  GeneAbundances %>% 
  filter(Type=="Unstratified" & GeneFamily!="UNMAPPED") %>%
  dplyr::select(SampleID, GeneFamily, RPKG) %>%
  pivot_wider(names_from = "SampleID", values_from = "RPKG", values_fill = 0) %>%
  pivot_longer(!GeneFamily, names_to = "SampleID", values_to="RPKG") %>%
  group_by(GeneFamily) %>%
  mutate(log2RPKG=log2(RPKG+(2/3*(min_nonzero(RPKG))))) %>%
  ungroup()
```


### Gene Diversity

```{r}
Gene_Shannon<-
  GeneAbundances_filled %>%
  dplyr::select(SampleID, GeneFamily, RPKG) %>%
  pivot_wider(names_from = SampleID, values_from = RPKG) %>%
  column_to_rownames("GeneFamily") %>%
  diversity(index="shannon", MARGIN=2)

Gene_Richness<-
  GeneAbundances_filled %>%
  dplyr::select(SampleID, GeneFamily, RPKG) %>%
  pivot_wider(names_from = SampleID, values_from = RPKG) %>%
  column_to_rownames("GeneFamily") %>%
  specnumber(MARGIN=2)

Gene_Diversity<-full_join(
  data.frame(Gene_Shannon) %>% rownames_to_column("SampleID"),
  data.frame(Gene_Richness) %>% rownames_to_column("SampleID")
)

interactive_table(Gene_Diversity)

Gene_Diversity %>%
  pivot_longer(!SampleID, names_to = "Metric", values_to = "Diversity") %>%
  left_join(metadata) %>%
  ggplot(aes(x=Group, y=Diversity, fill=Group)) +
  geom_boxplot(outlier.alpha = 0) +
  geom_jitter(height=0, width=0.2, shape=21) +
  facet_wrap(~Metric, scales="free")
ggsave("figures/Gene_diversity.pdf", height=7.5, width=10, useDingbats=F)


Gene_Diversity %>%
  pivot_longer(!SampleID, names_to = "Metric", values_to = "Diversity") %>%
  left_join(metadata) %>%
  group_by(Metric) %>%
  do(
    t.test(Diversity~Group, data=.) %>%
      broom::tidy()
  ) %>%
  interactive_table()
```

### Gene Ordination

```{r}
GeneDist<-
  GeneAbundances_filled %>%
  dplyr::select(SampleID, GeneFamily, RPKG) %>%
  pivot_wider(names_from = SampleID, values_from = RPKG) %>%
  column_to_rownames("GeneFamily") %>%
  t() %>%
  vegdist(., method="bray")

TaxPC<-ape::pcoa(GeneDist)

TaxPC$vectors %>%
  as.data.frame() %>%
  rownames_to_column("SampleID") %>%
  left_join(metadata) %>%
  ggplot(aes(x=Axis.1, y=Axis.2, fill=Group, label=SampleID)) +
  geom_point(shape=21) +
  ggrepel::geom_text_repel(size=3) +
  xlab(paste0("PC1:", round(TaxPC$values$Rel_corr_eig[1]*100, 2), "%")) +
  ylab(paste0("PC2:", round(TaxPC$values$Rel_corr_eig[2]*100, 2), "%"))

ggsave("figures/Gene_BrayPCoA.pdf", height=7.5, width=10, useDingbats=F)
```


### Gene Differential Abundance

Note: this may be a very long calculation (~30-60 min due to number of cases being tested)

```{r, eval=FALSE}
Gene_differential<-
  GeneAbundances_filled %>%
  #head(n=100000) %>%
  left_join(metadata) %>%
  group_by(GeneFamily) %>%
  do(
    t.test(log2RPKG~Group, data=.) %>%
      broom::tidy()
  )

Gene_differential<-
  Gene_differential %>%
  ungroup() %>%
  mutate(FDR=p.adjust(p.value, method="BH")) %>%
  mutate(ConfInterval95=paste0("[", round(conf.low,3) , " - ", round(conf.high,3), "]")) %>%
  dplyr::select(GeneFamily, log2FoldChange=estimate, Group1_mean=estimate1, Group2_mean=estimate2, Pvalue=p.value, FDR, ConfInterval95)


Gene_differential %>%
  mutate(Significant=if_else(FDR<0.1, "*","ns")) %>%
  mutate(Label=if_else(Significant=="*", GeneFamily, "")) %>%
  ggplot(aes(x=log2FC, y=-log10(Pvalue), color=Significant, label=GeneFamily)) +
  geom_point(shape=16, alpha=0.7) +
  ggrepel::geom_text_repel(size=3) +
  scale_color_manual(values=c("indianred","grey70")) 

ggsave("figures/Pathway_Volcano.pdf", height=7.5, width=10, useDingbats=F)


write_tsv(Gene_differential, "processed_data/differential_geneabundance.tsv")
#xfun::embed_file("processed_data/differential_geneabundance.tsv") Too large
```
